### PR TITLE
Fix for partial signals in old Django and old Python versions.

### LINKED ISSUE #1640: Django Signals integration breaks on partial objects for python <3.10
### How do you use Sentry?

Self-hosted/on-premise

### Version

1.9.9

### Steps to Reproduce

1. Use python older than 3.10.
2. Register a partial function as a signal handler.

### Expected Result

Signal is traced correctly.

### Actual Result

Exception is raised from `_get_receiver_name` function as `partial` objects don't have `__module__` before python 3.10 (and even there it's undocumented from what I can see).

It fails in our tests where we don't even register any signals so either Django itself or some kind of integration (Sentry?) registers such signals by default.

The whole signals integration is missing a `capture_internal_exceptions` context too I believe.

### PR DESCRIPTION
**Pull request by antonpirker**
Making sure signal name can also be set when partials are used as signal in  old Django versions in old Python versions.

Fixes #1640

### CODE REVIEW

File: sentry_sdk/integrations/django/signals_handlers.py [hunk 1]
```diff
@@ -21,9 +21,14 @@ def patch_signals():
 
     def _get_receiver_name(receiver):
         # type: (Callable[..., Any]) -> str
-        name = receiver.__module__ + "."
+        if hasattr(receiver, "__module__"):
```
Comment (sl0thentr0py) [inline sentry_sdk/integrations/django/signals_handlers.py:24]: can't we just use [`__qualname__`](https://docs.python.org/3/glossary.html#term-qualified-name) here?

Comment (antonpirker) [reply sentry_sdk/integrations/django/signals_handlers.py:24]: Good idea.

But I checked, the partials also do not have a `__qualname__`, so: no.

Comment (antonpirker) [reply sentry_sdk/integrations/django/signals_handlers.py:24]: oh, but we can use `__qualname__` if there is one, and if not we use `__name__`

**Review by sl0thentr0py — CHANGES_REQUESTED** [review]
also please add a test using a partial :)

File: sentry_sdk/integrations/django/signals_handlers.py [hunk 2]
```diff
@@ -13,19 +13,32 @@
...
+        name += receiver.__qualname__
+    elif hasattr(receiver, "__name__"):  # Python 2.7 has no __qualname__
+        name += receiver.__name__
+
+    if (
+        name == ""
+    ):  # certain functions (like partials) dont have a name so return the string representation
+        return str(receiver)
+
+    if hasattr(receiver, "__module__"):  # prepend with module, if there is one
```
Comment (sl0thentr0py) [inline sentry_sdk/integrations/django/signals_handlers.py:30]: qualname already has the module prefix right?

Comment (antonpirker) [reply sentry_sdk/integrations/django/signals_handlers.py:30]: No. It is only the fully qualified name of the class itself (with dots if is a nested class) but it does not include the python module.

Comment (sl0thentr0py) [reply sentry_sdk/integrations/django/signals_handlers.py:30]: ah gotcha, thx

**Review by sl0thentr0py — APPROVED** [review]

**Pull request merged**